consent-ui/src/app/ais/register/register.component.spec.ts

```diff
@@ -32,11 +35,54 @@ describe('RegisterComponent', () => {
 
   beforeEach(() => {
     fixture = TestBed.createComponent(RegisterComponent);
+    authService = fixture.debugElement.injector.get(AuthService);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+  it('should call registerUser in AuthService', () => {
+    authServiceSpy = spyOn(authService, 'userRegister').and.callThrough();
+    const form = component.registerForm;
+    form.controls['login'].setValue('alex');
+    form.controls['password'].setValue('1234');
+    component.onSubmit();
+    fixture.detectChanges();
+    expect(authServiceSpy).toHaveBeenCalledWith({ login: 'alex', password: '1234' });
+  });
+  it('should be true if the passwords are not same', () => {
+    authServiceSpy = spyOn(authService, 'userRegister').and.callThrough();
+    const form = component.registerForm;
+    form.controls['login'].setValue('alex');
+    form.controls['password'].setValue('adf');
+    form.controls['confirmPassword'].setValue('asdf');
+    component.onSubmit();
+    fixture.detectChanges();
+
+    expect(component.registerForm.invalid).toBe(true);
+  });
+  it('should be false if password is not set', () => {
+    authServiceSpy = spyOn(authService, 'userRegister').and.callThrough();
+    const form = component.registerForm;
+    form.controls['login'].setValue('alex');
+    form.controls['password'].setValue('');
+    form.controls['confirmPassword'].setValue('');
+    component.onSubmit();
+    fixture.detectChanges();
+
+    expect(component.registerForm.invalid).toBe(true);
+  });
+  it('should be false if username is not set', () => {
+    authServiceSpy = spyOn(authService, 'userRegister').and.callThrough();
+    const form = component.registerForm;
+    form.controls['login'].setValue('');
+    form.controls['password'].setValue('asdf');
```

Comment:
For the same passwords, it is better to use variable/constant instead of hardcoding - easier to understand 'sameness' fact